theano/scan_module/scan_perform.pyx

```diff
@@ -514,7 +514,8 @@ def perform(
             if i == 0:
                 jout = j+offset_out
                 shape = (store_steps[j],) + output_storage[jout].storage[0].shape
-                if len(output_storage[jout].storage[0].shape) == 0:
+                if output_storage[jout].storage[0].ndim == 0 and \
+                    isinstance(output_storage[jout].storage[0], numpy.ndarray):
```

Comment:
I made some changes to this PR, removed runtime checking and added compile time checking. Did some basic tests and found that only removing the 2 lines will result in errors.